src/librustc_mir/interpret/place.rs

```diff
@@ -1038,23 +1038,23 @@ where
         variant_index: VariantIdx,
         dest: PlaceTy<'tcx, M::PointerTag>,
     ) -> InterpResult<'tcx> {
-        let variant_scalar = Scalar::from_u32(variant_index.as_u32()).into();
+        if dest.layout.for_variant(self, variant_index).abi.is_uninhabited() {
```

Comment:
Please add a comment explaining what eddyb explained to me: that layout computation entirely excludes `is_uninhabited` variants from consideration, so that there just isn't a way to even represent their discriminant.